security/fma-kona-asterisc.md

```diff
@@ -0,0 +1,213 @@
+  Low impact since by default `kona` + `asterisc` are placed in the cold path.
+  [The "Kona and Asterisc Hot Path Failure"](#-Kona-and-Asterisc-Hot-Path-Failure) above covers the failure mode for `kona` + `asterisc` when placed in the hot path.
+
+- **Mitigations:**
+  Action tests covering the derivation pipeline are run on every pull request in the [kona] repository.
+  VM Runner exectutes the derivation pipeline as part of the kona fault proof program running ontop of asterisc.
+  [Extensive test coverage](https://app.codecov.io/gh/anton-rs/kona/tree/main/crates) of kona.
+
+- **Detection:**
+  The VM runner would pick up an issue.
+  The action tests or local testing may pick up an issue.
+  Use of the derivation elsewhere, for example in alternative consensus clients like `op-rs`, would pick up an issue.
+
+- **Recovery Path(s):**
+  Fix the derivation pipeline divergence, release and deploy the updated kona + asterisc when in the coldpath.
+  If `kona` + `asterisc` is placed in the hotpath, either the `op-program` + `cannon` would need to be falled back on otherwise fallback to the permissed dispute game.
+  [The "Kona and Asterisc Hot Path Failure"](#-Kona-and-Asterisc-Hot-Path-Failure) above covers the failure mode for `kona` + `asterisc` when placed in the hot path.
+
+
+### Kona Execution Diverges from op-program
+
+- **Description:**
+  The Kona Fault Proof Program executes blocks in a stateless L2 block executor that uses optimism-revm as a backend.
+  If optimism-revm or the L2 block executor diverge in behavior from the `op-program`, the `kona` + `asterisc` proof system will be invalid.
+
+- **Risk Assessment:**
+  Low likelihood, low impact.
+  A small divergence in execution behavior can cause invalid game results.
+  But since kona + asterisc is in the coldpath, it doesn't have impact on anything in the production system.
+
+- **Mitigations:**
+  State testing against op-revm.
+  Action tests covering the L2 block executor are run on every pull request in the [kona] repository.
+  `op-reth` which uses the same op-execution environment as `kona`.
+
+- **Detection:**
+  Running `op-reth` so it's synced and following tip would provide a way to watch for chain splits.
+  A chain split occuring means `kona`'s execution (the same as `op-reth`) diverges from the `op-program`.
+
+- **Recovery Path(s):**
+  If `kona` + `asterisc` is in the coldpath, it would need to be fixed and re-deployed.
+  If `op-program` + `cannon` cannot be placed in the hotpath, and `kona` + `asterisc` fails, the permissioned fallback must be engaged.
+  [The "Kona and Asterisc Hot Path Failure"](#-Kona-and-Asterisc-Hot-Path-Failure) above covers the failure mode for `kona` + `asterisc` when placed in the hot path.
+
+
+### Breaking Backwards Compatibility for the op-challenger
+
+- **Description:**
+  The `op-challenger` has been updated to support the Kona + Asterisc Fault Dispute Game type ([Game Type 3]).
```

Comment:
For personal knowledge what is the difference between `GameType 3` and `GameType 2` described here contains Asterisc in the name causing me to be confuse compare to cannon only having one gameType: 
https://github.com/ethereum-optimism/optimism/blob/39dc07942350899bc93b413a1688efd17048982f/op-challenger/game/fault/types/types.go#L29-L30

Reply:
`AsteriscGameType` = `2` = `op-program` + `asterisc`
`AsteriscKonaGameType` = `3` = `kona` + `asterisc`